src/simulationMachine.tsx

```diff
@@ -303,7 +303,10 @@ export const simulationMachine = simModel.createMachine(
             (ctx, e) => {
               return {
                 type: 'xstate.event',
-                event: e.event,
+                event: {
+                  ...e.event,
+                  origin: e.event.origin || ctx.currentSessionId,
+                },
```

Comment:
I don't think this is necessarily the correct thing to do. In my mind sending events from the viz resembles this piece of code:
```js
const service = interpret(machine.start)

onClick={() => service.send(createEvent())}
```
And in this scenario, the `origin` **wont** be set: https://codesandbox.io/s/hopeful-elion-0l4vg?file=/src/index.js

So, from my perspective, in this basic and most common scenario this wont resemble the production usage and thus it might introduce confusing behavior.

I recognize how not setting `origin` in some other scenarios might be "incorrect" as well though and it's something I've been mentioning to David some time ago. We've concluded though that, for the most part, this should be a less common scenario and that we won't bother with it at the moment. 

When this can be incorrect? When an event can only be received from another machine in the real application and when one uses the `origin` to do something (like responding to it). We can't currently determine possible sources though so we don't have a way to provide any hints about them (so they could be selected or something).

Keep in mind that sending events from the Viz to any non-root machine is often risky - in a sense that it might make the whole thing behave differently than in a production use case (unless your application literally is sending directly to such a machine, which is a valid use case). In a lot of scenarios, you could end up with some inconsistent state that way. One that would not be achievable by the real app.

Reply:
One of those scenarios in which not setting the origin is a breaking behavior is when we send a custom event to an external inspected source. How would we solve that without setting the origin to the current interpreted machine? Maybe conditionally doing it only if we're in the inspector mode?

Reply:
Could we go through this case on a call on Monday? I would love to get a little bit more context about this and this would allow me to jump into this quicker

Reply:
I would probably recommend dropping the `toSCXMLEvent` conversion from here: https://github.com/statelyai/xstate-viz/blob/7cc57c836c111eb1695a76b02de8f3d689cf59a6/src/EventsPanel.tsx#L388

and doing that here:
```js
toSCXMLEvent(e.event, { origin: ctx.currentSessionId })
```